src/app/pipes/date-format.pipe.ts

```diff
@@ -7,18 +7,22 @@ import * as moment from 'moment';
 })
 export class DateFormatPipe implements PipeTransform {
 
-  defaultDateFormat: string;
-
   constructor(private settingsService: SettingsService) {
-    this.defaultDateFormat = this.settingsService.dateFormat;
-    this.defaultDateFormat = this.defaultDateFormat.replace('dd', 'DD');
   }
 
-  transform(value: Date | moment.Moment, dateFormat: string): any {
+  transform(value: any, dateFormat: string): any {
+    const defaultDateFormat = this.settingsService.dateFormat.replace('dd', 'DD');
+    if (typeof value === 'undefined') {
+      return "";
+    }
+    let dateVal = moment(value);
+    if (value instanceof Array) {
+      dateVal = moment(value.join("-"), "YYYY-MM-DD");
+    }
     if (dateFormat == null) {
-      return moment(value).format(this.defaultDateFormat);
+      return moment(dateVal).format(defaultDateFormat);
```

Comment:
moment(dateVal) looks unnecessary to me... it is already a Moment object by this stage:
`return dateVal.format(this.defaultDateFormat);` is enough...i think